tap_wikipedia/wikipedia_abstracts_stream.py

```diff
@@ -116,16 +206,110 @@ def get_records(self, context: Dict | None) -> Iterable[Dict]:
         # get Wikipedia records
         records = self.__get_wikipedia_records(cached_file_path)
 
-        # returns a list of featured Wikipedia Article records
-        def get_featured_records() -> Iterable[Dict]:
+        # generates a stream of featured Wikipedia article records
+        def get_featured_records(records) -> Iterable[Dict]:
             featured_urls = self.__get_featured_articles()
 
             for record in records:
                 if record["abstract_info"]["url"] in featured_urls:
                     yield record
 
-        # Filter out featured Wikipedia Article records
-        if "featured" in subset_specification:
-            yield from get_featured_records()
-        else:
-            yield from records
+        # adds the main image of the Wikipedia article to the Wikipedia record
+        def add_images_to_records(records) -> Iterable[Dict]:
+            for record in records:
+                try:
+                    img_url = self.__get_wikipedia_record_images(
+                        record["abstract_info"]["url"]
+                    )
+                except Exception:
+                    self.__logger.warning(
+                        f'error getting URL of {record["abstract_info"]["title"]}',  # noqa: E501
+                        exc_info=True,
+                    )
+                    continue
+
+                record["abstract_info"]["image"] = img_url
+                yield record
```

Comment:
In my taps I keep things as Pydantic models right up until the point I want to yield them from get_records, then transform to a JSON-compatible dict with `.dict()`.

Reply:
Yes, this is much better. The records are now being kept as models until the very end.